### PR TITLE
Allow tracks to modify their width and height

### PR DESCRIPTION
**Pull request by alexander-veit**
## Description

> What was changed in this pull request?

This PR enables tracks to modify their height and width by publishing 'trackDimensionsModified'.

> Why is it necessary?

This is necessary for the Transcripts track which needs to adjust its own height when the number of shown transcripts changes.

Fixes #___

## Checklist

- [x] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [x] Unit tests added or updated
- [ ] Documentation added or updated
- [ ] Example(s) added or updated
- [ ] Update schema.json if there are changes to the viewconf JSON structure format
- [ ] Screenshot for visual changes (e.g. new tracks or UI changes)
- [ ] Updated CHANGELOG.md

### CODE REVIEW

File: app/scripts/HiGlassComponent.js [hunk 1]
```diff
@@ -4415,6 +4422,33 @@ class HiGlassComponent extends React.Component {
     );
   }
 
+  /**
+   * Handle trackDimensionsModified events
+   * settings.trackId = id of the track
+   * settings.height = new height of the track or undefined if current height should remain
+   * settings.width = new width of the track or undefined if current width should remain
+   */
+  trackDimensionsModifiedHandler(settings) {
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassComponent.js:4431]: Track uids are not guaranteed to be globally unique. You need a view / track uid pair to uniquely identify a track.

Comment (alexander-veit) [reply app/scripts/HiGlassComponent.js:4431]: Oh yes, of course. Thanks. Is there a simple way to get the current viewId from within a (plugin) track?

Comment (pkerpedjiev) [reply app/scripts/HiGlassComponent.js:4431]: Unfortunately, I don't think there is. You could perhaps add it to the context here: https://github.com/higlass/higlass/blob/9ceffd59cd75dff315cf67e33185da185b8c7419/app/scripts/TrackRenderer.js#L1491 but I'm not sure that it's even present in TrackRenderer.

Another possibility is to pass a callback into the track object that lets it change its size. See the viewport projection tracks as an example of this:

https://github.com/higlass/higlass/blob/9ceffd59cd75dff315cf67e33185da185b8c7419/app/scripts/TrackRenderer.js#L1621

Comment (alexander-veit) [reply app/scripts/HiGlassComponent.js:4432]: Thanks. It turns out the TrackRenderer knows about it. It just wasn't used so far.

**Pull request merged**